packages/@react-aria/interactions/test/useInteractOutside.test.js

```diff
@@ -206,3 +207,238 @@ describe('useInteractOutside', function () {
     });
   });
 });
+
+describe('useInteractOutside (iframes)', function () {
+  let iframe;
+  let iframeRoot;
+  let iframeDocument;
+  beforeEach(() => {
+    iframe = document.createElement('iframe');
+    window.document.body.appendChild(iframe);
+    iframeDocument = iframe.contentWindow.document;
+    iframeRoot = iframeDocument.createElement('div');
+    iframeDocument.body.appendChild(iframeRoot);
+  });
+
+  afterEach(() => {
+    iframe.remove();
+  });
+
+  const IframeExample = (props) => {
+    React.useEffect(() => {
+      ReactDOMRender(<Example {...props} />, iframeRoot);
+    }, [props]);
+
+    return null;
+  };
+
+  // TODO: JSDOM doesn't yet support pointer events. Once they do, convert these tests.
+  // https://github.com/jsdom/jsdom/issues/2527
+  describe('pointer events', function () {
+    installPointerEvent();
+
+    it('should fire interact outside events based on pointer events', async function () {
+      let onInteractOutside = jest.fn();
+      render(
+        <IframeExample onInteractOutside={onInteractOutside} />
+      );
+
+      await waitFor(() => {
+        expect(document.querySelector('iframe').contentWindow.document.body.querySelector('div[data-testid="example"]')).toBeTruthy();
+      });
+
+      const el = document.querySelector('iframe').contentWindow.document.body.querySelector('div[data-testid="example"]');
+      fireEvent(el, pointerEvent('pointerdown'));
+      fireEvent(el, pointerEvent('pointerup'));
+      expect(onInteractOutside).not.toHaveBeenCalled();
+
+      fireEvent(iframeDocument.body, pointerEvent('pointerdown'));
+      fireEvent(iframeDocument.body, pointerEvent('pointerup'));
+      expect(onInteractOutside).toHaveBeenCalledTimes(1);
```

Comment:
what should happen if there is a click outside of this iframe?

a more complicated question, these tests don't appear to mimic the setup that you've described at stripe where you have a central controlling iframe with multiple other iframes you render into from that central one
I feel like we should cover that in the tests as well, with clicks in any of the iframes or on the main document as well as just in the target

Reply:
@snowystinger we are actually mirroring the behavior here, just that instead of portalling from one iframe to another, we are portalling from the main window to an iframe. Without using the `getOwnerDocument`, the tests would refer to the main window, i.e. the main iframe in Stripe's use case, which would be incorrect.

Reply:
ah, ok, so the main controlling iframe contains all the other iframes as children, it can't reach out to any other siblings

and then for the other question, what happens if a click occurs outside the child iframe?

Reply:
Ah so for that we have observers on every iframe, i.e. if a click happens in another frame, we use `postMessage` to tell all other frames that a click occurred and unmount the component manually ourselves